client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs

```diff
@@ -69,11 +69,14 @@ public static class AutoRegistrationInMonitoring
                     if (monitoringRegistration.Url == myMonitoringUrl)
                         return;
 
-                    log.WriteMonitor("Auto-registration in monitoring", podTag, $"There is a registration for {myMonitoringName} in monitoring service!");
+                    if (monitoringRegistration.Url != _missingEnvVarUrl)
+                    {
+                        log.WriteMonitor("Auto-registration in monitoring", podTag, $"There is a registration for {myMonitoringName} in monitoring service!");
 
-                    myMonitoringUrl = _missingEnvVarUrl;
-                    string instanceTag = string.IsNullOrEmpty(podTag) ? Guid.NewGuid().ToString() : podTag;
-                    myMonitoringName = $"{myMonitoringName}-{instanceTag}";
+                        myMonitoringUrl = _missingEnvVarUrl;
+                        string instanceTag = string.IsNullOrEmpty(podTag) ? Guid.NewGuid().ToString() : podTag;
+                        myMonitoringName = $"{myMonitoringName}-{instanceTag}";
+                    }
                 }
                 catch (HttpOperationException)
                 {
```

Comment:
Why do we need this catch (or why do we interpret it such a way) if duplicated registration is detected on the line 72?

Reply:
Service throws HttpOperationException with 404 when registration with such service name is not found